aws/src/main/scala/com/gu/contentapi/client/IAMSigner.scala

```diff
@@ -24,14 +26,13 @@ class IAMSigner(credentialsProvider: AWSCredentialsProvider, awsRegion: String)
     * Returns the given set of headers, updated to include AWS sig4v signed headers based on the request and the credentials
     *
     * @param headers  Current set of request headers
-    * @param url      Request URL, including params
+    * @param uri      Request URI, including params
     * @return         Updated set of headers, including the authorisation headers
     */
-  def addIAMHeaders(headers: Map[String, String], url: String): Map[String, String] = {
+  def addIAMHeaders(headers: Map[String, String], uri: URI): Map[String, String] = {
```

Comment:
Just thinking out loud here: do you even need `headers`? I don't know if the signing process reads existing headers, but if not then I would discard it.

Reply:
unfortunately yes, it needs the headers and the full url